ribbon/sign.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#----------------------------------------------------------------------------
+# Created By: Steven (steven@ribbon.finance)
+# Created Date: 04/04/2022
+# version ='0.01'
+# ---------------------------------------------------------------------------
+''' Module to generate signature to bid '''
+# ---------------------------------------------------------------------------
+
+# ---------------------------------------------------------------------------
+# Imports
+# ---------------------------------------------------------------------------
+import eth_keys
+from encode import TypedDataEncoder
+
+# ---------------------------------------------------------------------------
+# Signature Generator
+# ---------------------------------------------------------------------------
+class SignatureGenerator:
+  '''
+  Object to generate bid signature
+
+  Args:
+      privateKey (str): Private key of the user in hex format with 0x prefix
+
+  Attributes:
+      signer (object): Instance of signer to generate signature
+  '''
+  def __init__(self, privateKey: str) -> None:
+    self.signer = eth_keys.keys.PrivateKey(bytes.fromhex(privateKey[2:]))
+
+  def signMessage(self, messageHash: str) -> dict:
+    '''Sign a hash message using the signer object
+
+    Args:
+        messageHash (str): Message to signed in hex format with 0x prefix
+
+    Returns:
+        signature (dict): Signature split into v, r, s components
+    '''
+    signature = self.signer.sign_msg_hash(bytes.fromhex(messageHash[2:]))
+
+    return {
+      'v': signature.v + 27,
+      'r': hex(signature.r),
+      's': hex(signature.s)
+    }
+
+  def _signTypedDataV4(self, domain: dict, types: dict, value: dict) -> str:
```

Comment:
Can we avoid asking for the types?

We're giving the interface here, so we can have again a default like

```
BID_TYPES = {
    'Bid': [
      {'name': 'swapId', 'type': 'uint256'},
      {'name': 'nonce', 'type': 'uint256'},
      {'name': 'signerWallet', 'type': 'address'},
      {'name': 'sellAmount', 'type': 'uint256'},
      {'name': 'buyAmount', 'type': 'uint256'},
      {'name': 'referrer', 'type': 'address'},
    ]
  }

...

  def _signTypedDataV4(self, domain: dict, value: dict, types: dict = BID_TYPES) -> str:
```

Reply:
Yea, good idea